tests/business/siaquerysetrunner_test.py

```diff
@@ -0,0 +1,228 @@
+        AnyWithEntries(
+            {
+                "type": "DALServiceError",
+                "value": "No working capabilities endpoint provided",
+            }
+        )
+    )
+    assert sentry_error["contexts"]["phase"] == {
+        "phase": "make_client",
+        "started_at": ANY_AWARE_DATETIME_STR,
+    }
+    assert sentry_error["tags"] == {
+        "flock": "test",
+        "business": "SIAQuerySetRunner",
+        "phase": "make_client",
+    }
+    assert sentry_error["user"] == {"username": "bot-mobu-tapuser"}
+
+    (sentry_transaction,) = sentry_items.transactions
+    assert sentry_transaction["transaction"] == "SIAQuerySetRunner - startup"
+
+
+@pytest.mark.asyncio
+async def test_failure(
+    client: AsyncClient,
+    respx_mock: respx.Router,
+    events: Events,
+    sentry_items: Captured,
+) -> None:
+    mock_gafaelfawr(respx_mock)
+    with patch.object(pyvo.dal, "SIA2Service") as mock:
+        mock.return_value.search.side_effect = [Exception("some error")]
+
+        r = await client.put(
+            "/mobu/flocks",
+            json={
+                "name": "test",
+                "count": 1,
+                "user_spec": {"username_prefix": "bot-mobu-testuser"},
+                "scopes": ["exec:notebook"],
+                "business": {"type": "SIAQuerySetRunner"},
+            },
+        )
+        assert r.status_code == 201
+
+        # Wait until we've finished at least one loop and check the results.
+        data = await wait_for_business(client, "bot-mobu-testuser1")
+        assert data["business"]["failure_count"] == 1
+
+    # Confirm Sentry errors
+    (sentry_error,) = sentry_items.errors
+    assert sentry_error["contexts"]["phase"] == {
+        "phase": "mobu.sia.search",
+        "started_at": ANY_AWARE_DATETIME_STR,
+    }
+    assert sentry_error["contexts"]["query_info"] == {
+        "started_at": ANY_AWARE_DATETIME_STR,
+        "query": AnySearch("SIA2Query"),
+    }
+    assert sentry_error["tags"] == {
+        "flock": "test",
+        "business": "SIAQuerySetRunner",
+        "phase": "mobu.sia.search",
+    }
+    assert sentry_error["user"] == {"username": "bot-mobu-testuser1"}
+
+    (sentry_transaction,) = sentry_items.transactions
+    assert sentry_transaction["transaction"] == "SIAQuerySetRunner - execute"
+
+    # Confirm metrics events
+    published = cast(MockEventPublisher, events.sia_query).published
+    published.assert_published_all(
+        [
+            {
+                "business": "SIAQuerySetRunner",
+                "duration": NOT_NONE,
+                "flock": "test",
+                "success": False,
+                "username": "bot-mobu-testuser1",
+            }
+        ]
+    )
+
+
+@pytest.mark.asyncio
+async def test_random_object(events: Events) -> None:
+    query_set = "dp02"
+    user = AuthenticatedUser(
+        username="bot-mobu-user", scopes=["read:tap"], token="blah blah"
```

Comment:
Is `read:tap` the correct scope that a bot user will need to access the SIA service? It doesn't matter in the test, but we'll have to be sure that the correct scope is configured in [the mobu values in phalanx](https://github.com/lsst-sqre/phalanx/blob/tickets%2F48587%2Fmobu-scale-test/applications/mobu/values-idfint.yaml#L48-L52)

Reply:
Good catch! It's actually `read:image` so I've corrected that bit accordingly